main.go

```diff
@@ -68,11 +51,34 @@ outer:
 		case <-dhcpEvents:
 			debounced.Reset(cfg.DebounceInterval)
 		case <-debounced.C:
-			l, err := leases.ReadLeases(cfg.LeaseFile)
+			ls, err := leases.ReadLeases(cfg.LeaseFile)
 			if err != nil {
 				log.Fatalw("could not parse leases file", "error", err)
 			}
-			err = r.Report(l)
+			active := ls.FilterActive()
+			byMac := active.LatestByMac()
+			log.Infow("reporting leases to metal-api", "all", len(ls), "active", len(active), "uniqueActive", len(byMac))
+
+			mtx := new(sync.Mutex)
+			var items []*leases.ReportItem
+
+			wg := new(sync.WaitGroup)
+			wg.Add(len(byMac))
+
+			for _, l := range byMac {
+				item := leases.NewReportItem(l, log)
+				go func() {
+					item.EnrichWithBMCDetails(cfg.IpmiPort, cfg.IpmiUser, cfg.IpmiPassword)
+					mtx.Lock()
+					items = append(items, item)
+					wg.Done()
+					mtx.Unlock()
+				}()
+			}
+
+			wg.Wait()
```

Comment:
Is it there some insurance that reports do not queue up when some of them are hanging? (like... in case a call to Redfish does not return)

Reply:
Such a context timeout configuration should be provided by go-hal. We can do this in an separate PR